lmnet/lmnet/post_processor.py

```diff
@@ -311,3 +313,106 @@ def __call__(self, outputs, **kwargs):
+            data_format (string): currently support only "NHWC".
+            compatible_tensorflow_v1 (bool): When the flag is True, it is compatible with tensorflow v1 resize function. Otherwise tensorflow v2 and Pillow resize function.
+                Tensorflow v1 image resize function `tf.image.resize_bilinear()` which has the bug of calculataion of center pixel. The bug is fixed in tensorflow v2 `tf.image.resize()` which given the same result as Pillow's resize.
+                See also https://github.com/tensorflow/tensorflow/issues/6720 and https://github.com/tensorflow/tensorflow/commit/3ae2c6691b7c6e0986d97b150c9283e5cc52c15f
+        """ # NOQA
+
+        self.size = size
+        self.data_format = data_format
+        # TODO(wakisaka): support "NCHW" format.
+        assert data_format == "NHWC", "data_format only support NHWC but given {}".format(data_format)
+        self.compatible_tensorflow_v1 = compatible_tensorflow_v1
+
+    def __call__(self, outputs, **kwargs):
+        """
+        Args:
+            outputs (numpy.ndarray): 4-D ndarray of network outputs to be resized cahnnel-wise.
+
+        Returns:
+            all args (dict):
+                outputs (numpy.ndarray): resized outputs. 4-D ndarray.
+        """
+        output_height = self.size[0]
+        output_width = self.size[1]
+
+        input_height = outputs.shape[0]
+        input_width = outputs.shape[1]
+
+        # TODO(wakisaka): should support downsample.
+        assert output_height >= input_height
+        assert output_width >= input_width
+
+        batch_size = len(outputs)
+        results = []
+        for i in range(batch_size):
+            image = outputs[i, :, :, :]
+            image = self._bilinear(image, size=self.size, compatible_tensorflow_v1=self.compatible_tensorflow_v1)
+            results.append(image)
+        results = np.array(results)
+        return dict({'outputs': results}, **kwargs)
+
+    @staticmethod
+    def _bilinear(inputs, size, compatible_tensorflow_v1=True):
+        output_height = size[0]
+        output_width = size[1]
+
+        input_height = inputs.shape[0]
+        input_width = inputs.shape[1]
+        channel = inputs.shape[2]
+
+        if compatible_tensorflow_v1:
+            scale = [(input_height - 1)/(output_height - 1), (input_width - 1)/(output_width - 1)]
+        else:
+            scale = [(input_height - 0)/(output_height - 0), (input_width - 0)/(output_width - 0)]
+
+        output = np.zeros((output_height, output_width, channel)).astype(inputs.dtype)
+
+        for h in range(output_height):
+            for w in range(output_width):
+                if compatible_tensorflow_v1:
+                    center_y = h * scale[0]
+                    center_x = w * scale[1]
+                else:
+                    center_y = (h + 0.5) * (scale[0]) - 0.5
+                    center_x = (w + 0.5) * (scale[1]) - 0.5
+
+                if compatible_tensorflow_v1:
```

Comment:
I think this `if` can be merged with `if` at line 386.

Reply:
sure

Reply:
Fixed in https://github.com/blue-oil/blueoil/pull/211/commits/3ad532032db1b916c93865f4384180b15f672601